### PR TITLE
B menu show butcher, disassemble and salvage times

### PR DESCRIPTION
**Pull request by OzoneH3**
#### Summary
```SUMMARY: Interface "B menu show butcher, disassemble and salvage times"```

#### Additional context
Salvage times should probably be increased a little bit...

Butcher Menu:

Cut up menu:

### CODE REVIEW

**Comment by AMurkin** [comment]
I hope it will be more structured like Consume item window.

consume

**Comment by OzoneH3** [comment]
Only if someone adds columns to the uimenu/uilist. Converting this to use the inventory_selector system would be quite a pita. But I agree it's not very pretty, especially the cut up lines.

File: src/game.cpp [hunk 1]
```diff
@@ -9387,18 +9415,22 @@ void game::butcher()
...
+            auto cut_time = [&]( const char *text, enum butcher_type bt ) {
+                return string_format( text, to_string_clipped( time_duration::from_turns( butcher_time_to_cut( u,
+                                      items[corpses[ret]], bt ) / 100 ) ) );
+            };
             uilist smenu;
             smenu.desc_enabled = true;
             smenu.text = _( "Choose type of butchery:" );
-            smenu.addentry_desc( BUTCHER, true, 'B', _( "Quick butchery" ),
+            smenu.addentry_desc( BUTCHER, true, 'B', cut_time( _( "Quick butchery (%s)" ), BUTCHER ),
                                  _( "This techinque is used when you are in a hurry, but still want to harvest something from the corpse.  Yields are lower as you don't try to be precise, but it's useful if you don't want to set up a workshop.  Prevents zombies from raising." ) );
```
Comment (Leland) [inline src/game.cpp]: Not yours but while you're here can you fix the spelling error?

"techinque" vs "technique"

**Comment by jbytheway** [comment]
I had been thinking that the "Butcher everything" options should clarify which type of butchering it will do.  So it could say e.g. "Quick butcher everything".  (Maybe that's always what it does; I'm not sure, hence the desire for clarification).  If you think it's appropriate on this PR, could you take a look at that?

**Comment by OzoneH3** [comment]
Changed the text to "Quick butcher everything".

**Comment by TranslucentVectors** [comment]
to OzoneH3 - Can I ask what font you use in game, the screenshots you have provided look so much clearer than any font I have tried to use.....it's such a visual uplift!

**Comment by AMurkin** [comment]
I think it's Ubuntu mono: https://fonts.google.com/specimen/Ubuntu+Mono

**Comment by TranslucentVectors** [comment]
Many thanks

File: src/activity_handlers.cpp [hunk 2]
```diff
@@ -412,6 +404,15 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
         }
     }
 
+    act.moves_left = butcher_time_to_cut( u, corpse_item, action );
+}
+
+int butcher_time_to_cut( const player &u, const item &corpse_item, const butcher_type action )
+{
+    const mtype &corpse = *( corpse_item.get_mtype() );
```
Comment (BevapDin) [inline src/activity_handlers.cpp]: Don't need the outer parentheses here.

File: src/activity_handlers.cpp [hunk 3]
```diff
@@ -412,6 +404,15 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
...
+    const int factor = u.max_quality( ( action == DISSECT ) ? quality_id( "CUT_FINE" ) :
```
Comment (BevapDin) [inline src/activity_handlers.cpp]: Nor here around the comparison.

File: src/game.cpp [hunk 4]
```diff
@@ -9108,9 +9109,11 @@ void add_corpses( uilist &menu, map_stack &items,
         hotkey = -1;
     }
 }
+
 // Salvagables stack so we need to pass in a stack vector rather than an item index vector
 void add_salvagables( uilist &menu, map_stack &items,
-                      const std::vector<std::pair<int, int>> &stacks, size_t &menu_index )
+                      const std::vector<std::pair<int, int>> &stacks, size_t &menu_index,
+                      const salvage_actor *salvage_iuse )
```
Comment (BevapDin) [inline src/game.cpp]: `salvage_iuse` should be a reference.

**Pull request merged**